Chessington.GameEngine/ExploreInOneDirection.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace Chessington.GameEngine.Tests.Pieces
+{
+    public class ExploreInOneDirection
+    {
+        public static IEnumerable<Square> ExploreOneDirection(Board board, Square initialSquare, int direction)
```

Comment:
I am not sure about this class tbh, let's have a chat in the morning if you didn't get passing Function as a parameter